lib/spack/spack/patch.py

```diff
@@ -119,7 +119,8 @@ def apply(self, stage):
             patch_stage.fetch()
             patch_stage.check()
             patch_stage.cache_local()
-            patch_stage.expand_archive()
+            if self.url.lower().endswith('.tar'):
+                patch_stage.expand_archive()
```

Comment:
@alalazo A naive question. Does the current implementation handle other archives? If expand_archive handles other archive types, e.g. `*.zip` it is simple to add. I just need to know what they are. Is this documented in the code?

Most patches are uncompressed as @adamjstewart had said.